{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "No index, query, or output file specified!\r\n",
      "Bowtie 2 version 2.2.4 by Ben Langmead (langmea@cs.jhu.edu, www.cs.jhu.edu/~langmea)\r\n",
      "Usage: \r\n",
      "  bowtie2 [options]* -x <bt2-idx> {-1 <m1> -2 <m2> | -U <r>} [-S <sam>]\r\n",
      "\r\n",
      "  <bt2-idx>  Index filename prefix (minus trailing .X.bt2).\r\n",
      "             NOTE: Bowtie 1 and Bowtie 2 indexes are not compatible.\r\n",
      "  <m1>       Files with #1 mates, paired with files in <m2>.\r\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\r\n",
      "  <m2>       Files with #2 mates, paired with files in <m1>.\r\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\r\n",
      "  <r>        Files with unpaired reads.\r\n",
      "             Could be gzip'ed (extension: .gz) or bzip2'ed (extension: .bz2).\r\n",
      "  <sam>      File for SAM output (default: stdout)\r\n",
      "\r\n",
      "  <m1>, <m2>, <r> can be comma-separated lists (no whitespace) and can be\r\n",
      "  specified many times.  E.g. '-U file1.fq,file2.fq -U file3.fq'.\r\n",
      "\r\n",
      "Options (defaults in parentheses):\r\n",
      "\r\n",
      " Input:\r\n",
      "  -q                 query input files are FASTQ .fq/.fastq (default)\r\n",
      "  --qseq             query input files are in Illumina's qseq format\r\n",
      "  -f                 query input files are (multi-)FASTA .fa/.mfa\r\n",
      "  -r                 query input files are raw one-sequence-per-line\r\n",
      "  -c                 <m1>, <m2>, <r> are sequences themselves, not files\r\n",
      "  -s/--skip <int>    skip the first <int> reads/pairs in the input (none)\r\n",
      "  -u/--upto <int>    stop after first <int> reads/pairs (no limit)\r\n",
      "  -5/--trim5 <int>   trim <int> bases from 5'/left end of reads (0)\r\n",
      "  -3/--trim3 <int>   trim <int> bases from 3'/right end of reads (0)\r\n",
      "  --phred33          qualities are Phred+33 (default)\r\n",
      "  --phred64          qualities are Phred+64\r\n",
      "  --int-quals        qualities encoded as space-delimited integers\r\n",
      "\r\n",
      " Presets:                 Same as:\r\n",
      "  For --end-to-end:\r\n",
      "   --very-fast            -D 5 -R 1 -N 0 -L 22 -i S,0,2.50\r\n",
      "   --fast                 -D 10 -R 2 -N 0 -L 22 -i S,0,2.50\r\n",
      "   --sensitive            -D 15 -R 2 -N 0 -L 22 -i S,1,1.15 (default)\r\n",
      "   --very-sensitive       -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\r\n",
      "\r\n",
      "  For --local:\r\n",
      "   --very-fast-local      -D 5 -R 1 -N 0 -L 25 -i S,1,2.00\r\n",
      "   --fast-local           -D 10 -R 2 -N 0 -L 22 -i S,1,1.75\r\n",
      "   --sensitive-local      -D 15 -R 2 -N 0 -L 20 -i S,1,0.75 (default)\r\n",
      "   --very-sensitive-local -D 20 -R 3 -N 0 -L 20 -i S,1,0.50\r\n",
      "\r\n",
      " Alignment:\r\n",
      "  -N <int>           max # mismatches in seed alignment; can be 0 or 1 (0)\r\n",
      "  -L <int>           length of seed substrings; must be >3, <32 (22)\r\n",
      "  -i <func>          interval between seed substrings w/r/t read len (S,1,1.15)\r\n",
      "  --n-ceil <func>    func for max # non-A/C/G/Ts permitted in aln (L,0,0.15)\r\n",
      "  --dpad <int>       include <int> extra ref chars on sides of DP table (15)\r\n",
      "  --gbar <int>       disallow gaps within <int> nucs of read extremes (4)\r\n",
      "  --ignore-quals     treat all quality values as 30 on Phred scale (off)\r\n",
      "  --nofw             do not align forward (original) version of read (off)\r\n",
      "  --norc             do not align reverse-complement version of read (off)\r\n",
      "  --no-1mm-upfront   do not allow 1 mismatch alignments before attempting to\r\n",
      "                     scan for the optimal seeded alignments\r\n",
      "  --end-to-end       entire read must align; no clipping (on)\r\n",
      "   OR\r\n",
      "  --local            local alignment; ends might be soft clipped (off)\r\n",
      "\r\n",
      " Scoring:\r\n",
      "  --ma <int>         match bonus (0 for --end-to-end, 2 for --local) \r\n",
      "  --mp <int>         max penalty for mismatch; lower qual = lower penalty (6)\r\n",
      "  --np <int>         penalty for non-A/C/G/Ts in read/ref (1)\r\n",
      "  --rdg <int>,<int>  read gap open, extend penalties (5,3)\r\n",
      "  --rfg <int>,<int>  reference gap open, extend penalties (5,3)\r\n",
      "  --score-min <func> min acceptable alignment score w/r/t read length\r\n",
      "                     (G,20,8 for local, L,-0.6,-0.6 for end-to-end)\r\n",
      "\r\n",
      " Reporting:\r\n",
      "  (default)          look for multiple alignments, report best, with MAPQ\r\n",
      "   OR\r\n",
      "  -k <int>           report up to <int> alns per read; MAPQ not meaningful\r\n",
      "   OR\r\n",
      "  -a/--all           report all alignments; very slow, MAPQ not meaningful\r\n",
      "\r\n",
      " Effort:\r\n",
      "  -D <int>           give up extending after <int> failed extends in a row (15)\r\n",
      "  -R <int>           for reads w/ repetitive seeds, try <int> sets of seeds (2)\r\n",
      "\r\n",
      " Paired-end:\r\n",
      "  -I/--minins <int>  minimum fragment length (0)\r\n",
      "  -X/--maxins <int>  maximum fragment length (500)\r\n",
      "  --fr/--rf/--ff     -1, -2 mates align fw/rev, rev/fw, fw/fw (--fr)\r\n",
      "  --no-mixed         suppress unpaired alignments for paired reads\r\n",
      "  --no-discordant    suppress discordant alignments for paired reads\r\n",
      "  --no-dovetail      not concordant when mates extend past each other\r\n",
      "  --no-contain       not concordant when one mate alignment contains other\r\n",
      "  --no-overlap       not concordant when mates overlap at all\r\n",
      "\r\n",
      " Output:\r\n",
      "  -t/--time          print wall-clock time taken by search phases\r\n",
      "  --un <path>           write unpaired reads that didn't align to <path>\r\n",
      "  --al <path>           write unpaired reads that aligned at least once to <path>\r\n",
      "  --un-conc <path>      write pairs that didn't align concordantly to <path>\r\n",
      "  --al-conc <path>      write pairs that aligned concordantly at least once to <path>\r\n",
      "  (Note: for --un, --al, --un-conc, or --al-conc, add '-gz' to the option name, e.g.\r\n",
      "  --un-gz <path>, to gzip compress output, or add '-bz2' to bzip2 compress output.)\r\n",
      "  --quiet            print nothing to stderr except serious errors\r\n",
      "  --met-file <path>  send metrics to file at <path> (off)\r\n",
      "  --met-stderr       send metrics to stderr (off)\r\n",
      "  --met <int>        report internal counters & metrics every <int> secs (1)\r\n",
      "  --no-unal          supppress SAM records for unaligned reads\r\n",
      "  --no-head          supppress header lines, i.e. lines starting with @\r\n",
      "  --no-sq            supppress @SQ header lines\r\n",
      "  --rg-id <text>     set read group id, reflected in @RG line and RG:Z: opt field\r\n",
      "  --rg <text>        add <text> (\"lab:value\") to @RG line of SAM header.\r\n",
      "                     Note: @RG line only printed when --rg-id is set.\r\n",
      "  --omit-sec-seq     put '*' in SEQ and QUAL fields for secondary alignments.\r\n",
      "\r\n",
      " Performance:\r\n",
      "  -p/--threads <int> number of alignment threads to launch (1)\r\n",
      "  --reorder          force SAM output order to match order of input reads\r\n",
      "  --mm               use memory-mapped I/O for index; many 'bowtie's can share\r\n",
      "\r\n",
      " Other:\r\n",
      "  --qc-filter        filter out reads that are bad according to QSEQ filter\r\n",
      "  --seed <int>       seed for random number generator (0)\r\n",
      "  --non-deterministic seed rand. gen. arbitrarily instead of using read attributes\r\n",
      "  --version          print version information and quit\r\n",
      "  -h/--help          print this usage message\r\n",
      "(ERR): bowtie2-align exited with value 1\r\n"
     ]
    }
   ],
   "source": [
    "!bowtie2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2-build \\\n",
    "/Users/sr320/git-repos/student-fish546-2016/data/Panopea_generosa-Scaff-10k.fa \\\n",
    "/Volumes/Alanine/wd/17-09-15/Panopea_generosa-Scaff-10k.bowtie-index \\\n",
    "&> /Volumes/Alanine/wd/17-09-15/bw-stderr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# RNA-seq Files"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false,
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "78147 reads; of these:\r\n",
      "  78147 (100.00%) were paired; of these:\r\n",
      "    65054 (83.25%) aligned concordantly 0 times\r\n",
      "    11972 (15.32%) aligned concordantly exactly 1 time\r\n",
      "    1121 (1.43%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    65054 pairs aligned concordantly 0 times; of these:\r\n",
      "      3682 (5.66%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    61372 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      122744 mates make up the pairs; of these:\r\n",
      "        117997 (96.13%) aligned 0 times\r\n",
      "        3805 (3.10%) aligned exactly 1 time\r\n",
      "        942 (0.77%) aligned >1 times\r\n",
      "24.50% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x /Volumes/Alanine/wd/17-09-15/Panopea_generosa-Scaff-10k.bowtie-index \\\n",
    "-1 /Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L001_R1_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L002_R1_001.fastq.gz \\\n",
    "-2 /Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L001_R2_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L002_R2_001.fastq.gz \\\n",
    "-p 4 \\\n",
    "-S /Volumes/Alanine/wd/17-09-15/NR005.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "6473873 reads; of these:\r\n",
      "  6473873 (100.00%) were paired; of these:\r\n",
      "    5523074 (85.31%) aligned concordantly 0 times\r\n",
      "    860977 (13.30%) aligned concordantly exactly 1 time\r\n",
      "    89822 (1.39%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    5523074 pairs aligned concordantly 0 times; of these:\r\n",
      "      283935 (5.14%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    5239139 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      10478278 mates make up the pairs; of these:\r\n",
      "        10096854 (96.36%) aligned 0 times\r\n",
      "        299721 (2.86%) aligned exactly 1 time\r\n",
      "        81703 (0.78%) aligned >1 times\r\n",
      "22.02% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x /Volumes/Alanine/wd/17-09-15/Panopea_generosa-Scaff-10k.bowtie-index \\\n",
    "-1 /Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L001_R1_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L002_R1_001.fastq.gz \\\n",
    "-2 /Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L001_R2_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L002_R2_001.fastq.gz \\\n",
    "-p 4 \\\n",
    "-S /Volumes/Alanine/wd/17-09-15/NR006.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "669693 reads; of these:\r\n",
      "  669693 (100.00%) were paired; of these:\r\n",
      "    647021 (96.61%) aligned concordantly 0 times\r\n",
      "    20100 (3.00%) aligned concordantly exactly 1 time\r\n",
      "    2572 (0.38%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    647021 pairs aligned concordantly 0 times; of these:\r\n",
      "      23245 (3.59%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    623776 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      1247552 mates make up the pairs; of these:\r\n",
      "        1105568 (88.62%) aligned 0 times\r\n",
      "        93044 (7.46%) aligned exactly 1 time\r\n",
      "        48940 (3.92%) aligned >1 times\r\n",
      "17.46% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x /Volumes/Alanine/wd/17-09-15/Panopea_generosa-Scaff-10k.bowtie-index \\\n",
    "-1 /Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L001_R1_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L002_R1_001.fastq.gz \\\n",
    "-2 /Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L001_R2_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L002_R2_001.fastq.gz \\\n",
    "-p 4 \\\n",
    "-S /Volumes/Alanine/wd/17-09-15/NR012.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "68487 reads; of these:\r\n",
      "  68487 (100.00%) were paired; of these:\r\n",
      "    67936 (99.20%) aligned concordantly 0 times\r\n",
      "    515 (0.75%) aligned concordantly exactly 1 time\r\n",
      "    36 (0.05%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    67936 pairs aligned concordantly 0 times; of these:\r\n",
      "      538 (0.79%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    67398 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      134796 mates make up the pairs; of these:\r\n",
      "        128942 (95.66%) aligned 0 times\r\n",
      "        4500 (3.34%) aligned exactly 1 time\r\n",
      "        1354 (1.00%) aligned >1 times\r\n",
      "5.86% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x /Volumes/Alanine/wd/17-09-15/Panopea_generosa-Scaff-10k.bowtie-index \\\n",
    "-1 /Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L001_R1_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L002_R1_001.fastq.gz \\\n",
    "-2 /Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L001_R2_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L002_R2_001.fastq.gz \\\n",
    "-p 4 \\\n",
    "-S /Volumes/Alanine/wd/17-09-15/NR019.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "7613170 reads; of these:\r\n",
      "  7613170 (100.00%) were paired; of these:\r\n",
      "    6464278 (84.91%) aligned concordantly 0 times\r\n",
      "    1024680 (13.46%) aligned concordantly exactly 1 time\r\n",
      "    124212 (1.63%) aligned concordantly >1 times\r\n",
      "    ----\r\n",
      "    6464278 pairs aligned concordantly 0 times; of these:\r\n",
      "      310801 (4.81%) aligned discordantly 1 time\r\n",
      "    ----\r\n",
      "    6153477 pairs aligned 0 times concordantly or discordantly; of these:\r\n",
      "      12306954 mates make up the pairs; of these:\r\n",
      "        11785504 (95.76%) aligned 0 times\r\n",
      "        392488 (3.19%) aligned exactly 1 time\r\n",
      "        128962 (1.05%) aligned >1 times\r\n",
      "22.60% overall alignment rate\r\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/bowtie2-2.2.4/bowtie2 \\\n",
    "-x /Volumes/Alanine/wd/17-09-15/Panopea_generosa-Scaff-10k.bowtie-index \\\n",
    "-1 /Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L001_R1_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L002_R1_001.fastq.gz \\\n",
    "-2 /Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L001_R2_001.fastq.gz,/Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L002_R2_001.fastq.gz \\\n",
    "-p 4 \\\n",
    "-S /Volumes/Alanine/wd/17-09-15/NR021.sam\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Volumes/Alanine/wd/17-09-15\n"
     ]
    }
   ],
   "source": [
    "cd /Volumes/Alanine/wd/17-09-15/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Started analysis of NR005_S4_L001_R1_001.fastq.gz\n",
      "Started analysis of NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Started analysis of NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Started analysis of NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 10% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Started analysis of NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 10% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 15% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Started analysis of NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 60% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 70% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 20% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 25% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 25% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 30% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR005_S4_L002_R1_001.fastq.gz\n",
      "Analysis complete for NR005_S4_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 45% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 50% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 55% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 60% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 65% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 70% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 55% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 60% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 65% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 70% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 75% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Analysis complete for NR005_S4_L002_R2_001.fastq.gz\n",
      "Approx 40% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Analysis complete for NR005_S4_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 60% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 35% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 70% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 80% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 90% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Analysis complete for NR006_S3_L001_R1_001.fastq.gz\n",
      "Analysis complete for NR006_S3_L002_R1_001.fastq.gz\n",
      "Started analysis of NR006_S3_L001_R2_001.fastq.gz\n",
      "Started analysis of NR006_S3_L002_R2_001.fastq.gz\n",
      "Started analysis of NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 10% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 15% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 20% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 5% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Started analysis of NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 30% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 30% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 75% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Started analysis of NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 5% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Analysis complete for NR012_S1_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 25% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 60% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Started analysis of NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 65% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 25% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 80% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 85% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 15% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 90% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 95% complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 55% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 30% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 60% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 70% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 80% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 85% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 45% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 95% complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Analysis complete for NR012_S1_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 50% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR012_S1_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 80% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 85% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 90% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 95% complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 55% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR012_S1_L002_R2_001.fastq.gz\n",
      "Approx 50% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Started analysis of NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 5% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 10% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 15% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 20% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 25% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 65% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 70% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 75% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 60% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Analysis complete for NR019_S7_L001_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 65% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 70% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 85% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 75% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 80% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 85% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Analysis complete for NR006_S3_L002_R2_001.fastq.gz\n",
      "Approx 90% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Started analysis of NR019_S7_L002_R1_001.fastq.gz\n",
      "Started analysis of NR019_S7_L001_R2_001.fastq.gz\n",
      "Started analysis of NR019_S7_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR006_S3_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 30% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 75% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 85% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 55% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 60% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 70% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Analysis complete for NR019_S7_L002_R1_001.fastq.gz\n",
      "Approx 75% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 5% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Started analysis of NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 15% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 85% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 90% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 25% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 95% complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 35% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 40% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 45% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 50% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 55% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 60% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 65% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 70% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 80% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR019_S7_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 90% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Approx 95% complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR019_S7_L002_R2_001.fastq.gz\n",
      "Started analysis of NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Started analysis of NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Started analysis of NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 20% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 35% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 40% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 5% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 45% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 5% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 10% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 55% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 20% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 60% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 25% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 65% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 30% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 35% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 70% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 10% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 15% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 75% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 45% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 80% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 50% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 85% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 55% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 90% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 95% complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 15% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 20% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR021_S8_L002_R1_001.fastq.gz\n",
      "Approx 65% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 70% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 20% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 85% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 25% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 95% complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Analysis complete for NR021_S8_L002_R2_001.fastq.gz\n",
      "Approx 25% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 30% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 30% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 35% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 35% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 40% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 40% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 45% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 45% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 50% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 50% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 70% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 80% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 85% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 90% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 95% complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 55% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Analysis complete for NR021_S8_L001_R1_001.fastq.gz\n",
      "Approx 60% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 65% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 70% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Too many tiles (>500) so giving up trying to do per-tile qualities since we're probably parsing the file wrongly\n",
      "Approx 75% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 80% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 85% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 90% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Approx 95% complete for NR021_S8_L001_R2_001.fastq.gz\n",
      "Analysis complete for NR021_S8_L001_R2_001.fastq.gz\n"
     ]
    }
   ],
   "source": [
    "!/Applications/bioinfo/FastQC/fastqc -t 6 \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L001_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L002_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L001_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR005_S4_L002_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L001_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L002_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L001_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR006_S3_L002_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L001_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L002_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L001_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR012_S1_L002_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L001_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L002_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L001_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR019_S7_L002_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L001_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L002_R1_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L001_R2_001.fastq.gz \\\n",
    "/Volumes/web/halfshell/working-directory/17-09-07/NR021_S8_L002_R2_001.fastq.gz \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[INFO   ]         multiqc : This is MultiQC v1.0.dev0\n",
      "[INFO   ]         multiqc : Template    : default\n",
      "[INFO   ]         multiqc : Searching '/Volumes/web/halfshell/working-directory/17-09-07/'\n",
      "[INFO   ]          fastqc : Found 20 reports\n",
      "[INFO   ]         multiqc : Report      : multiqc_report.html\n",
      "[INFO   ]         multiqc : Data        : multiqc_data\n",
      "[INFO   ]         multiqc : MultiQC complete\n"
     ]
    }
   ],
   "source": [
    "!multiqc /Volumes/web/halfshell/working-directory/17-09-07/\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
